### PR TITLE
[instrument_list] BVL QC panel bug fix- can't save the QC status and QC type properly.  

### PR DESCRIPTION
**Pull request by kongtiaowang**
This pull request fixes the bug that can't save the null value and can't save the data into SQL table in some cases. It also fixes the logic error.

### CODE REVIEW

File: modules/instrument_list/php/instrument_list_controlpanel.class.inc [hunk 1]
```diff
@@ -280,27 +280,25 @@ class Instrument_List_ControlPanel extends \TimePoint
 
         // set the BVLQC flag
         $user =& \User::singleton();
-        if (isset($_REQUEST['setBVLQCStatus'])
+        if (isset($_REQUEST['setBVLQCStatus']) && !empty($_REQUEST['setBVLQCStatus'])
```
Comment (zaliqarosli) [inline modules/instrument_list/php/instrument_list_controlpanel.class.inc]: I believe `empty()` already checks for whether a variable exists so `isset` here is not neccesary?

File: modules/instrument_list/php/instrument_list_controlpanel.class.inc [hunk 2]
```diff
@@ -280,27 +280,25 @@ class Instrument_List_ControlPanel extends \TimePoint
...
         ) {
-            $status = \Utility::nullifyEmpty(
-                $_REQUEST['setBVLQCStatus'],
-                'BVLQCStatus'
-            );
-            $this->setData('BVLQCStatus', $status);
+            $this->setData('BVLQCStatus', $_REQUEST['setBVLQCStatus']);
+            return;
+        }
+        if (isset($_REQUEST['setBVLQCStatus']) && empty($_REQUEST['setBVLQCStatus'])
```
Comment (zaliqarosli) [inline modules/instrument_list/php/instrument_list_controlpanel.class.inc]: similarly, if it is not set, empty will return false

**Comment by kongtiaowang** [comment]
@zaliqarosli good catch!
It is done.

File: modules/instrument_list/php/instrument_list_controlpanel.class.inc [hunk 3]
```diff
@@ -280,24 +280,22 @@ class Instrument_List_ControlPanel extends \TimePoint
...
         ) {
-            $QCtype = \Utility::nullifyEmpty(
-                $_REQUEST['setBVLQCType'],
-                'BVLQCType'
-            );
-            $this->setData('BVLQCType', $QCtype);
+            $this->setData('BVLQCStatus', null);
+            return;
+        }
+        if (!empty($_REQUEST['setBVLQCType'])
```
Comment (zaliqarosli) [inline modules/instrument_list/php/instrument_list_controlpanel.class.inc]: could these 4 if statements be condensed into 2 if/else blocks?

Comment (kongtiaowang) [reply modules/instrument_list/php/instrument_list_controlpanel.class.inc]: @ZainVirani  the first if statement needs to check "$user->hasPermission('bvl_feedback')". 
the second if statement does not need to check this permission.

Comment (kongtiaowang) [reply modules/instrument_list/php/instrument_list_controlpanel.class.inc]: @zaliqarosli  you are right!

**Pull request merged**